modules/javafx.graphics/src/main/java/com/sun/glass/ui/headless/HeadlessRobot.java

```diff
@@ -0,0 +1,406 @@
+            if (oldWindow != null) {
+                HeadlessView oldView = (HeadlessView)oldWindow.getView();
+                if (oldView != null) {
+                    int owx = oldWindow.getX();
+                    int owy = oldWindow.getY();
+                    oldView.notifyMouse(MouseEvent.EXIT, MouseEvent.BUTTON_NONE, (int)mouseX-owx, (int)mouseY-owy, (int)mouseX, (int)mouseY, modifiers, true, true);
+                }
+            }
+        }
+    }
+
+    private HeadlessWindow getTopWindow() {
+        List<Window> windows = Window.getWindows().stream()
+                .filter(win -> win.getView() != null)
+                .filter(win -> !win.isClosed())
+                .filter(win -> !win.isMinimized()).toList();
+        if (windows.isEmpty()) return null;
+        if (windows.size() == 1) return (HeadlessWindow)windows.get(0);
+        return (HeadlessWindow)windows.get(windows.size() -1);
+    }
+
+    private HeadlessWindow getFocusedWindow() {
+        List<Window> windows = Window.getWindows().stream()
+                .filter(win -> win.getView()!= null)
+                .filter(win -> !win.isClosed())
+                .filter(win -> win.isFocused()).toList();
+        if (windows.isEmpty()) return null;
+        if (windows.size() == 1) return (HeadlessWindow)windows.get(0);
+        return (HeadlessWindow)windows.get(windows.size() -1);
+    }
+
+    private HeadlessWindow getTargetWindow(double x, double y) {
+        List<Window> windows = Window.getWindows().stream()
+                .filter(win -> win.getView()!= null)
+                .filter(win -> !win.isClosed())
+                .filter(win -> (x >= win.getX() && x <= win.getX() + win.getWidth()
+                        && y >= win.getY() && y <= win.getY()+ win.getHeight())).toList();
+        if (windows.isEmpty()) {
+            return null;
+        }
+        if (windows.size() == 1) return (HeadlessWindow)windows.get(0);
+        return (HeadlessWindow)windows.get(windows.size() -1);
+    }
+
+    int getModifiers(MouseButton... buttons) {
+        int modifiers = KeyEvent.MODIFIER_NONE;
+        for (int i = 0; i < buttons.length; i++) {
+            switch (buttons[i]) {
+                case PRIMARY:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_PRIMARY;
+                    break;
+                case MIDDLE:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_MIDDLE;
+                    break;
+                case SECONDARY:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_SECONDARY;
+                    break;
+                case BACK:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_BACK;
+                    break;
+                case FORWARD:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_FORWARD;
+                    break;
+            }
```

Comment:
Using an expression can help increase readibility and correctness:
```suggestion
            modifiers |= switch (buttons[i]) {
                case NONE -> KeyEvent.MODIFIER_NONE;
                case PRIMARY -> KeyEvent.MODIFIER_BUTTON_PRIMARY;
                case MIDDLE -> KeyEvent.MODIFIER_BUTTON_MIDDLE;
                case SECONDARY -> KeyEvent.MODIFIER_BUTTON_SECONDARY;
                case BACK -> KeyEvent.MODIFIER_BUTTON_BACK;
                case FORWARD -> KeyEvent.MODIFIER_BUTTON_FORWARD;
            };
```

Reply:
suggestion is merged